CMakeLists.txt

```diff
@@ -18,10 +18,30 @@ endif()
 
 enable_testing()
 
-# Leatherman setup
-list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/vendor/leatherman/cmake")
+include(FeatureSummary)
+
+SET(LEATHERMAN_COMPONENTS locale catch nowide logging util file_util dynamic_library execution ruby rapidjson)
+
+# We look for curl early, because whether or not we link to the leatherman curl library
+# is dependent on whether or not we find curl on the system.
+if ((("${CMAKE_SYSTEM_NAME}" MATCHES "Linux") OR WIN32) AND NOT WITHOUT_CURL)
```

Comment:
Restricting to Linux or WIN32 doesn't make sense for all projects, just Facter (where those are the only OS we expect to run on EC2 or GCE). I'd change it to just `NOT WITHOUT_CURL`.